polkadot/node/primitives/src/lib.rs

```diff
@@ -498,6 +498,11 @@ pub struct CollationGenerationConfig {
 	pub collator: Option<CollatorFn>,
 	/// The parachain that this collator collates for
 	pub para_id: ParaId,
+	/// Whether elastic scaling is enabled for this collation.
+	/// If it is, the collator will send the parent-head data along with the collation.
+	/// If your parachain is not assigned to multiple cores at the same time,
+	/// you should set this to `false`.
+	pub with_elastic_scaling: bool,
```

Comment:
Can we not detect if the connected nodes support `v3` of the networking protocol and if they do, we send directly the new version that includes the head? 

This feels like some hack here to me. Just always request the parent head from the collation generation and then internally decide if you send this to the validator or not.

Reply:
IIUC, we only want to send the parent head data only for paras that actually use elastic scaling. Or do you think it should be the default (although seems a bit wasteful)? Can add it as `v3` is there's a consensus on this.

In addition,
> Add an additional (backwards compatibility) advertise collation message

not sure what meant by backwards compatibility here. In the current implementation validators are required to upgrade before collators, but I don't think it's a big deal.

Reply:
Wouldn't it be a more simpler and elegant approach to add a specific protocol for just fetching the heads? Something like `ParaHeadFetching`. The validator knows if the parachain has multiple cores assigned to it so it will send  both `CollationFetchingV2` and `ParaHeadFetching` in parallel if that is the case.

Reply:
> Wouldn't it be a more simpler and elegant approach to add a specific protocol for just fetching the heads?

Sounds good to me.

Reply:
I went with the approach suggested by @eskimor [here](https://github.com/paritytech/polkadot-sdk/pull/3302#discussion_r1487504082). Any objections to that?

Reply:
> I went with the approach suggested by @eskimor [here](https://github.com/paritytech/polkadot-sdk/pull/3302#discussion_r1487504082). Any objections to that?

None, that's also fine